app/cdap/components/PipelineDiff/util/reactflowGraph.ts

```diff
@@ -0,0 +1,113 @@
+import { PLUGIN_TYPES } from 'components/hydrator/components/Canvas/constants';
+import {
+  generatePluginMapKey,
+  getCustomIconSrc,
+  pluginDiffColors,
+  shouldShowAlertsPort,
+  shouldShowErrorsPort,
+} from './helpers';
+import { getConnectionDiffKey, getStageDiffKey } from './diff';
+
+/**
+ * Given a pipeline configuration, constructs a pipeline graph that can be displayed
+ * in a reacflow canvas.
+ * @param config a pipeline's configuration
+ * @returns nodes and connections to be displayed in a reactflow pipeline graph
+ */
+export function getReactflowPipelineGraph(
+  config: IPipelineConfig | null,
+  availablePluginsMap: any,
+  diffMap: IPipelineDiffMap
+) {
+  if (!config) {
+    return { nodes: [], connections: [] };
+  }
+  const graphNodes = config.stages.map((stage) => {
+    return {
+      name: stage.name,
+      type: stage.plugin.type,
+      label: stage.plugin.label,
+    };
+  });
+  const graphConnections = config.connections;
+  const graph = getGraphLayout(graphNodes, graphConnections, 200);
+
+  const stageNameToStage: { [stageName: string]: IPipelineStage } = {};
+
+  const nodes = config.stages.map((stage) => {
+    stageNameToStage[stage.name] = stage;
+    const pluginMapKey = generatePluginMapKey(stage);
+
+    const hasErrorAndAlert =
+      shouldShowAlertsPort(availablePluginsMap, pluginMapKey) ||
+      shouldShowErrorsPort(availablePluginsMap, pluginMapKey);
+    const type: 'alertErrorNode' | 'defaultNode' = hasErrorAndAlert
+      ? 'alertErrorNode'
+      : 'defaultNode';
+    const diffItem = diffMap.stages[getStageDiffKey(stage)];
+
+    return {
+      id: stage.name,
+      data: {
+        ...stage,
+        customIconSrc: getCustomIconSrc(availablePluginsMap, pluginMapKey),
+        iconName: getPluginIcon(stage.plugin.name),
+        diffItem,
+      },
+      type,
+      position: {
+        x: graph._nodes[stage.name].x,
+        y: graph._nodes[stage.name].y,
+      },
+    };
+  });
+
+  const connections = config.connections.map((connection) => {
+    const getStageDiffKeyFromPluginName = (name: string) => getStageDiffKey(stageNameToStage[name]);
+    const connectionKey = getConnectionDiffKey(connection, getStageDiffKeyFromPluginName);
+    const diffIndicator =
+      objectQuery(diffMap, 'connections', connectionKey, 'diffIndicator') ?? 'default';
+
+    let sourceHandle = 'source_right';
+    if (stageNameToStage[connection.to].plugin.type === PLUGIN_TYPES.ERROR_TRANSFORM) {
+      sourceHandle = 'source_error';
+    } else if (stageNameToStage[connection.to].plugin.type === PLUGIN_TYPES.ALERT_PUBLISHER) {
+      sourceHandle = 'source_alert';
+    }
+
+    return {
+      id: `reactflow__edge-${connection.from}-${connection.to}`,
+      source: connection.from,
+      target: connection.to,
+      sourceHandle,
+      type: 'smoothstep',
+      isSelectable: true,
+      markerEnd: {
+        type: MarkerType.ArrowClosed,
+        color: pluginDiffColors[diffIndicator] + '88',
```

Comment:
you're adding 88 to the color here?

Reply:
It was for me to easily add alpha value to the color. Converted the object into a function and all needed colors are part of an object returned by the function.